test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/RazorEngineTest.cs

```diff
@@ -0,0 +1,219 @@
+        }
+
+        [Fact]
+        public void RazorEngine_InjectWithSemicolon_Runtime()
+        {
+            RunRuntimeTest("InjectWithSemicolon");
+        }
+
+        [Fact]
+        public void RazorEngine_Model_Runtime()
+        {
+            RunRuntimeTest("Model");
+        }
+
+        [Fact]
+        public void RazorEngine_ModelExpressionTagHelper_Runtime()
+        {
+            RunRuntimeTest("ModelExpressionTagHelper");
+        }
+        #endregion
+
+        #region DesignTime
+        [Fact]
+        public void RazorEngine_Basic_DesignTime()
+        {
+            RunDesignTimeTest("Basic");
+        }
+
+        [Fact]
+        public void RazorEngine_ViewImports_DesignTime()
+        {
+            RunDesignTimeTest("_ViewImports");
+        }
+
+        [Fact]
+        public void RazorEngine_Inject_DesignTime()
+        {
+            RunDesignTimeTest("Inject");
+        }
+
+        [Fact]
+        public void RazorEngine_InjectWithModel_DesignTime()
+        {
+            RunDesignTimeTest("InjectWithModel");
+        }
+
+        [Fact]
+        public void RazorEngine_InjectWithSemicolon_DesignTime()
+        {
+            RunDesignTimeTest("InjectWithSemicolon");
+        }
+
+        [Fact]
+        public void RazorEngine_Model_DesignTime()
+        {
+            RunDesignTimeTest("Model");
+        }
+
+        [Fact]
+        public void RazorEngine_MultipleModels_DesignTime()
+        {
+            RunDesignTimeTest("MultipleModels");
+        }
+
+        [Fact]
+        public void RazorEngine_ModelExpressionTagHelper_DesignTime()
+        {
+            RunDesignTimeTest("ModelExpressionTagHelper");
+        }
+        #endregion
+
+        private static void RunRuntimeTest(string testName)
+        {
+            // Arrange
+            var inputFile = "TestFiles/Input/" + testName + ".cshtml";
+            var outputFile = "TestFiles/Output/Runtime/" + testName + ".cs";
+            var expectedCode = ResourceFile.ReadResource(_assembly, outputFile, sourceFile: false);
+
+            var engine = RazorEngine.Create(b =>
+            {
+                InjectDirective.Register(b);
+                ModelDirective.Register(b);
+
+                b.Features.Add(new ModelExpressionPass());
+                b.Features.Add(new MvcViewDocumentClassifierPass());
+                b.Features.Add(new DefaultInstrumentationPass());
+
+                b.Features.Add(new Microsoft.CodeAnalysis.Razor.DefaultTagHelperFeature());
+
+                var references = AppDomain.CurrentDomain
+                    .GetAssemblies()
+                    .Select(assembly => MetadataReference.CreateFromFile(assembly.Location))
+                    .ToList<MetadataReference>();
+
+                b.Features.Add(new Microsoft.CodeAnalysis.Razor.DefaultMetadataReferenceFeature()
+                {
+                    References = references,
+                });
+            });
```

Comment:
We should refactor this into a static method somewhere in Mvc.Razor.Host so we can don't have it hardcoded in tests.